HW04/aleksei_zhuchkov/lib/mentor.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Mentor
+  attr_accessor :storage_students
+  attr_reader :name, :surname
+
+  def initialize(mentor)
+    @name = mentor[:name].capitalize
+    @surname = mentor[:surname].capitalize
+    @storage_students = []
+  end
+
+  def add_homework(info_homework)
+    homework = Homework.new(info_homework[:title], info_homework[:description], info_homework[:student])
+    info_homework[:student].add_homework_to_student_storage(homework)
+    info_homework[:student].storage_notification_for_student << Notification.new('Add homework', homework.title)
+    homework
+  end
+
+  def subscribe_to!(student)
+    storage_students << student
+  end
+
+  def notifications
+    notification_message = Notification.new
+    notification_message.print_notification_for_students_and_mentor(name, surname, storage_students, 'mentor')
+  end
+
+  def mark_as_read!
+    storage_students.each { |student| student.storage_notification_for_mentor.clear }
```

Comment:
This is a mentor class, mentor should not read students notifications, he should clear only his own notifications.